### PR TITLE
C++: Revert `semmle.code.cpp.dataflow` to its old state

### PR DESCRIPTION
**Pull request by jketema**
While here make sure all queries and tests use IR dataflow when appropriate.

This basically restores the situation where we have "dataflow" and "ir.datatflow". Next step create a "dataflow.new" which points to "ir.dataflow". Do we also want to remove the dataflow library copy from experimental and redirect that to "ir.dataflow" (or even just remove it without redirection)?

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll
Comment (jketema) [inline cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll:2]: I made this private, as there's no guarantee that the user of this library uses the same dataflow library. Alternative would be to make a copy of the library for ir dataflow.

Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll:2]: Does this need a change note?

File: cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll
Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll:4]: Does this need a change note?

File: cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll
Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll:2]: Well spotted! Indeed, I don't think any library should publicly export such dependencies. This technically is a breaking change (although a very trivial one for users to fix if they forgot to import the dataflow library in their own queries), so we should probably add a change note.

File: cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll
Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll:4]: Yeah, I think we can lump all these changes together under a single change note.

**Review by MathiasVP — APPROVED** [review]
A single comment, but otherwise this LGTM if CI is happy (I'm heavily basing my review on the fact that I'd expect CI to complain if we forgot to rewrite one of the imports 🤞). Maybe we shouldn't merge this until the other CI issues has been fixed? 😂

**Comment by MathiasVP** [comment]
> Do we also want to remove the dataflow library copy from experimental and redirect that to "ir.dataflow" (or even just remove it without redirection)?

I think we should just remove it without redirection.

File: cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll
Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll:2]: Done.

File: cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll
Comment (jketema) [reply cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll:4]: Done.

**Comment by jketema** [comment]
> I think we should just remove it without redirection

Will do in a separate PR.

**Comment by MathiasVP** [comment]
Feel free to ignore the QLDoc CI failure on the old dataflow library. I don't see any value in adding QLDoc to these given that we're about to deprecate them 🤞.

**Comment by jketema** [comment]
> Feel free to ignore the QLDoc CI failure on the old dataflow library. I don't see any value in adding QLDoc to these given that we're about to deprecate them 🤞.

That was the plan

**Pull request merged**
